worlds/checksmate/__init__.py

```diff
@@ -0,0 +1,212 @@
+from typing import List, Dict, ClassVar, Type
+
+from BaseClasses import Tutorial, Region, MultiWorld, Item, CollectionState
+from Options import PerGameCommonOptions
+from worlds.AutoWorld import WebWorld, World
+
+from .Options import CMOptions
+from .Items import CMItem, item_table, item_name_groups
+from .Locations import CMLocation, location_table, Tactic
+from .Presets import checksmate_option_presets
+from .Rules import set_rules
+from .CollectionState import CMCollectionState
+from .ItemPool import CMItemPool
+from .PieceModel import PieceModel, PieceLimitCascade
+from .MaterialModel import MaterialModel
+
+
+class CMWeb(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the ChecksMate software on your computer. This guide covers single-player, "
+        "multiworld, and related software.",
+        "English",
+        "checksmate_en.md",
+        "checks-mate/en",
+        ["roty", "rft50"]
+    )]
+
+    options_presets = checksmate_option_presets
+
+
+class CMWorld(World):
+    """
+    ChecksMate is a game where you play chess, but all of your pieces were scattered across the multiworld.
+    You win when you checkmate the opposing king!
+    """
+    game: ClassVar[str] = "ChecksMate"
+    data_version = 0
+    web = CMWeb()
+    required_client_version = (0, 3, 0)
+    options_dataclass: ClassVar[Type[PerGameCommonOptions]] = CMOptions
+    options: CMOptions
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = {name: data.code for name, data in location_table.items()}
+    locked_locations: List[str]
+
+    item_name_groups = item_name_groups
+    items_used: Dict[int, Dict[str, int]] = {}
+    items_remaining: Dict[int, Dict[str, int]] = {}
+    armies: Dict[int, List[int]] = {}
```

Comment:
```suggestion
    items_used: ClassVar[Dict[int, Dict[str, int]]] = {}
    items_remaining: ClassVar[Dict[int, Dict[str, int]]] = {}
    armies: ClassVar[Dict[int, List[int]]] = {}
```
explicitly mark these as class vars (or make them instance vars that isn't per player)